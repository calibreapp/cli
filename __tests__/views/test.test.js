import formatTest from '../../src/views/markdown'

import completedTest from '../fixtures/completedTest.json'
import erroredTest from '../fixtures/erroredTest.json'
import timeoutTest from '../fixtures/timeoutTest.json'
import incompleteTest from '../fixtures/incompleteTest.json'

describe('test', () => {
  it('completed', () => {
    const formattedTest = formatTest(completedTest.markdownReport)
    expect(formattedTest).toMatchSnapshot()
  })

  it('errored', () => {
    const formattedTest = formatTest(erroredTest.markdownReport)
    expect(formattedTest).toMatchSnapshot()
  })

  it('timeout', () => {
    const formattedTest = formatTest(timeoutTest.markdownReport)
    expect(formattedTest).toMatchSnapshot()
  })

  it('incomplete', () => {
    const formattedTest = formatTest(incompleteTest.markdownReport)
    expect(formattedTest).toMatchSnapshot()
  })
})
