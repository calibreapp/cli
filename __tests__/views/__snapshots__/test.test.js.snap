// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test completed 1`] = `
"Performance report for your page. Learn more about Calibre's Single Page Tests ↗ (https://calibreapp.com/docs/automation/single-page-tests)

### Test summary

┌──────────────────────────────────────────────┬────────────────────────────────────────┬─────────────────┬──────────────────┬──────────────────┐
│ Tested Page                                  │ Test Status                            │ Test Location   │ Device Emulation │ Connection Speed │
├──────────────────────────────────────────────┼────────────────────────────────────────┼─────────────────┼──────────────────┼──────────────────┤
│ Inspect tested page (https://calibreapp.com) │ ✅ Completed (Inspect test (test-url)) │ 💻 Local laptop │ Chrome Desktop   │ Cable            │
└──────────────────────────────────────────────┴────────────────────────────────────────┴─────────────────┴──────────────────┴──────────────────┘

### Core Web Vitals

  Core Web Vitals assessment failed. Learn more about Core Web Vitals ↗ (https://calibreapp.com/blog/core-web-vitals)

┌──────────────────────────┬─────────────────────────┬─────────────────────┐
│ Largest Contentful Paint │ Cumulative Layout Shift │ Total Blocking Time │
├──────────────────────────┼─────────────────────────┼─────────────────────┤
│ 3748 sec (⚠️ To improve) │ 16 (✅ Good)            │ 2742 sec (❌ Poor)  │
└──────────────────────────┴─────────────────────────┴─────────────────────┘

### Assets transferred

┌─────────────────────────┬────────┐
│ Metric                  │ Value  │
├─────────────────────────┼────────┤
│ Total Js Transferred    │ 93 KB  │
├─────────────────────────┼────────┤
│ Total Html Transferred  │ 27 KB  │
├─────────────────────────┼────────┤
│ Total Image Transferred │ 16 KB  │
├─────────────────────────┼────────┤
│ Total Css Transferred   │ 74 KB  │
├─────────────────────────┼────────┤
│ Total Font Transferred  │ 62 KB  │
├─────────────────────────┼────────┤
│ Total Video Transferred │ 55 KB  │
├─────────────────────────┼────────┤
│ Total Json Transferred  │ 71 KB  │
├─────────────────────────┼────────┤
│ Total Transferred       │ 400 KB │
└─────────────────────────┴────────┘

"
`;

exports[`test errored 1`] = `
"## Error markdown here

"
`;

exports[`test incomplete 1`] = `
"Test in progress

"
`;

exports[`test timeout 1`] = `
"## Timeout markdown here

"
`;
